### DISCUSSION TITLE
Cannot get it to connect when using a namespace

### DISCUSSION THREAD

**Discussion by raquelhortab** [miguelgrinberg/Flask-SocketIO#1980] [Q&A]
Hi, I've already spent way too many hours on this and I'm starting to get crazy. I got an old project (that somehow used to work) but I can't get it to work locally. 

On the server side, this is the socket code:
```
socketio = SocketIO(app, message_queue="redis://") # I have tried with cors_allowed_origins="*" as well
socketio.on_namespace(GameNamespace(redis, '/game'))
```

installed packages:
```
Flask-SocketIO==2.8.1
python-socketio==1.6.1
socketIO-client==0.7.2
websocket-client==0.40.0
gevent-websocket==0.9.5
```

This is running on a docker container, run with `-p 3002:80`

On the client side, while this:
```
socket = io.connect('http://localhost:3002');
```
seems to connect the client to the server ("Client has connected to the server!" is printed on the console), I am not connected to the /game namespace so I messages don't really go anywhere.

If I try this, so it connects to the `/game` namespace:
```
socket = io.connect('http://localhost:3002/game');
```

I get this error on the JS console:
```
socket.io-1.7.3.min.js:2          GET http://localhost/socket.io/?EIO=3&transport=polling&t=OUyl-ld net::ERR_CONNECTION_REFUSED
```
It seems to be removing the port number. The thing it, it must have worked at some point and I believe I have not changed any versions (I've read it could be to version incompatibility)

Client-side library: `socket.io-1.7.3`

Thanks in advance

**Comment by miguelgrinberg** [comment]
You should use:

    socket = io.connect('http://localhost:3002/game');

**Reply by raquelhortab** [reply]
yes, obviously, I'm sorry, that's what I was using... I missed the port when writting, I'm gonna edit the post

**Reply by raquelhortab** [reply]
I tried again just in case (since I've tried many things already), the error I am getting with this config is:

```
Failed to load resource: the server responded with a status of 500 (Internal Server Error)
:3002/socket.io/?EIO=3&transport=polling&t=OUy_tox&sid=eff7001164ce4601a591cd1743fe0b81:1          
Failed to load resource: the server responded with a status of 400 (BAD REQUEST)
```

**Reply by miguelgrinberg** [reply]
Now you need to look at your server to learn more about the 500 and 400 errors it is returning to the client.

**Reply by raquelhortab** [reply]
Well I kinda fixed it. I still have no idea why it just stopped working. It turns out it had something to do with `request.referrer` from flask not containing the full url. That's where I got my room id to call join_room, and do some more things. Thanks a lot anyway, sorry for bothering!
